src/avram/nested_save_operation.cr

```diff
@@ -1,52 +1,79 @@
 module Avram::NestedSaveOperation
-  macro has_one(type_declaration)
-    {% name = type_declaration.var %}
-    {% type = type_declaration.type.resolve %}
-
-    {% model_type = type.ancestors.find do |t|
-         t.stringify.starts_with?("Avram::SaveOperation(")
-       end.type_vars.first %}
-
-    {% assoc = T.constant(:ASSOCIATIONS).find do |assoc|
-         assoc[:relationship_type] == :has_one &&
-           assoc[:type].resolve.name == model_type.name
-       end %}
-
-    {% unless assoc %}
-      {% raise "#{T} must have a has_one association with #{model_type}" %}
-    {% end %}
-
-    after_save save_{{ name }}
-
-    def save_{{ name }}(saved_record)
-      unless {{ name }}.save
-        add_error(:{{ name }}, "failed")
-        mark_nested_save_operations_as_failed
-        database.rollback
-      end
+  macro included
+    macro inherited
+      define_has_one
     end
+  end
+
+  macro define_has_one
+    macro has_one(type_declaration)
+      \{% name = type_declaration.var %}
+      \{% type = type_declaration.type.resolve %}
+
+      \{% model_type = type.ancestors.find do |t|
+          t.stringify.starts_with?("Avram::SaveOperation(")
+        end.type_vars.first %}
+
+      \{% assoc = T.constant(:ASSOCIATIONS).find do |assoc|
+          assoc[:relationship_type] == :has_one &&
+            assoc[:type].resolve.name == model_type.name
+        end %}
+
+      \{% if ! assoc %}
+        \{% raise "#{T} must have a has_one association with #{model_type}" %}
+      \{% end %}
 
-    def {{ name }}
-      @{{ name }} ||= if new_record?
-        {{ type }}.new(params)
-      else
-        {{ type }}.new(record.not_nil!.{{ assoc[:assoc_name].id }}!, params)
+      \{% for need in type.constant(:OPERATION_NEEDS) %}
+        needs \{{ name }}_\{{ need }}
```

Comment:
Not sure about this, but needed to forward the needs to the nested operation somehow.

Reply:
I think we should back out these changes. It'll be something broken, but we can open up an issue and tackle it for a future PR. My thought is you'd pass it in like `SaveBusiness.create(save_tax_id: {...})`, but there's some other complicated things that need to be figured out first.